src/github/client/github-client.ts

```diff
@@ -1,29 +1,46 @@
+import Logger from 'bunyan';
 import { Octokit } from "@octokit/rest";
 import axios, { AxiosInstance, AxiosRequestConfig, AxiosResponse } from "axios";
 import AppTokenHolder from "./app-token-holder";
 import InstallationTokenCache from "./installation-token-cache";
 import AuthToken from "./auth-token";
 import { GetPullRequestParams } from "./types";
-	
+import { handleFailedRequest, instrumentFailedRequest, instrumentRequest, setRequestStartTime } from "./interceptors";
+import { metricHttpRequest } from "../../config/metric-names";
+import { getLogger } from "../../config/logger";
+import {urlParamsMiddleware} from "../../util/axios/common-middleware";
+
 /**
  * A GitHub client that supports authentication as a GitHub app.
  *
  * @see https://docs.github.com/en/developers/apps/building-github-apps/authenticating-with-github-apps
  */
 export default class GitHubClient {
-
 	private readonly axios: AxiosInstance;
 	private readonly appTokenHolder: AppTokenHolder;
 	private readonly installationTokenCache: InstallationTokenCache;
 	private readonly githubInstallationId: number;
+	private logger: Logger;
 
 	constructor(
 		githubInstallationId: number,
+		logger: Logger,
 		baseURL = "https://api.github.com"
 	) {
+		this.logger = logger || getLogger("github.client.axios");
 		this.axios = axios.create({
 			baseURL
 		});
+		this.axios.interceptors.request.use(urlParamsMiddleware)
+		this.axios.interceptors.request.use(setRequestStartTime);
+		this.axios.interceptors.response.use(
+			undefined,
+			handleFailedRequest(this.logger)
```

Comment:
Is there no better way to add an error handler with Axios? I remember stumbling across this when building the initial version of the `GitHubClient`, but I couldn't find a better solution, either. Passing `undefined` is a code smell for me.

Reply:
I couldnt find any better way then this. I hoped that they would have taken an array of interceptors but could find any version that did that.